crates/bevy_ui/src/render/ui_material_pipeline.rs

```diff
@@ -731,9 +731,12 @@ pub fn queue_ui_material_nodes<M: UiMaterial>(
 {
     let draw_function = draw_functions.read().id::<DrawUiMaterial<M>>();
 
-    for (entity, extracted_uinode) in extracted_uinodes.uinodes.iter() {
-        let material = render_materials.get(&extracted_uinode.material).unwrap();
-        for (view, mut transparent_phase) in &mut views {
+    for (view, mut transparent_phase) in &mut views {
+        transparent_phase
+            .items
+            .reserve(extracted_uinodes.uinodes.len());
+        for (entity, extracted_uinode) in extracted_uinodes.uinodes.iter() {
+            let material = render_materials.get(&extracted_uinode.material).unwrap();
```

Comment:
This means the material will have to be looked up once per view per entity instead of once per entity